hooks/settings.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2014 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk import TankError
+import os
+import re
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+class ExportSettings(HookBaseClass):
+    """
+    This hook controls the settings that flame will use when it exports the quicktimes 
+    prior to uploading them to Shotgun. It also lets a user control where on disk temporary
+    quicktime files will be located.
+    """
+
+    def get_export_preset(self):
+        """
+        Return the path to a flame export preset that should be used when generating 
+        a sequence quicktime.
+        
+        :returns: Path on disk to flame export preset 
+        """
+        return self._generate_profile_xml()
+
+    ###############################################################################################
+    # helper methods and internals
+    
+    def _write_content_to_file(self, content, file_name):
+        """
+        Helper method. Writes content to file and returns the path.
+        The content will be written to the app specific cache location 
+        on disk, organized by app instance name. The rationale is that 
+        each app instance holds its own configuration, and the configuration
+        generates one set of unique xml files.
+        
+        :param content: Data to write to the file
+        :param file_name: The name of the file to create
+        :returns: path to the created file
+        """
+        # determine location
+        file_path = os.path.join(self.parent.cache_location, self.parent.instance_name, file_name)
+        folder = os.path.dirname(file_path)
```

Comment:
Seems a bit weird to build a full file path, including its folder, and then to extract the folder name from it, if I may. Why not doing two os.path.join, the second one being `file_path = os.path.join(folder, file_name)` ?